expression/expression.go

```diff
@@ -218,7 +218,8 @@ func evalExprToString(expr Expression, row []types.Datum, _ *variable.StatementC
 	if val.IsNull() || err != nil {
 		return res, val.IsNull(), errors.Trace(err)
 	}
-	if expr.GetTypeClass() == types.ClassString || IsHybridType(expr) {
+	exprEvalTp := fieldTp2EvalTp(expr.GetType())
+	if exprEvalTp == tpString || exprEvalTp == tpJSON || IsHybridType(expr) {
```

Comment:
why not check tpDatetime, tpTimestamp here?

Reply:
it should be checked in evalTime